app/src/test/java/ch/epfl/sdp/TrapTest.java

```diff
@@ -0,0 +1,65 @@
+package ch.epfl.sdp;
+
+import org.junit.Test;
+
+import ch.epfl.sdp.entity.EntityType;
+import ch.epfl.sdp.entity.Player;
+import ch.epfl.sdp.entity.PlayerManager;
+import ch.epfl.sdp.game.Game;
+import ch.epfl.sdp.item.ItemBox;
+import ch.epfl.sdp.item.Trap;
+import ch.epfl.sdp.map.GeoPoint;
+import ch.epfl.sdp.map.MapsActivity;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+
+public class TrapTest {
+    @Test
+    public void trapCanBeSetUpAndDoesDamage() throws InterruptedException {
+        Player owner = new Player(45, 45, 100, "username1", "email1@email.com");
+        Player opponent = new Player(39, 39, 100, "username2", "email2@email.com");
+
+        MockMapApi mockMapApi = new MockMapApi();
+        MapsActivity.setMapApi(mockMapApi);
+        mockMapApi.setCurrentLocation(owner.getLocation());
+
+        Game game = new Game();
+        game.initGame();
+

```

Comment:
You can do this setup part in a separate method annotated with @Before

Reply:
Since I only have one test, I thought it was not useful for now, but I will do it.